src/components/UploadForm.js

```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react'
+import ProgressBar from './ProgressBar';
+import "../styles/upload-form.css"
+
+const UploadForm = () => {
+  
+
+    const [ file, setFile ] = useState(null);
+    const [ error, setError ] = useState(null);
+    
+    const types = ['image/png', 'image/jpeg']
+
+    const changeHandler = (e) => { 
+        let selected = e.target.files[0];
+        console.log(selected)
+        if(selected && types.includes(selected.type)) { 
+            setFile(selected)
+            setError('')
+        } else {
+            setFile(null)
+            setError('Please select an acceptable image type (PNG or JPEG)')
+        }
+    };
+  return (
+    <form id="upload-form">
+        <input type="file" onChange={changeHandler}></input>
+    
+        <div>
+            <div className='output'>
+                { error && <div className='error'>{error}</div>}
+                { file && <div>{file.name}</div>}
+                { file && <ProgressBar file={file} setFile={setFile}/> }
+            </div>
```

Comment:
You could consider a ternary here, to make sure your state is exhaustive. 
Right now, this works, but the way it is conditionally rendered could be bug prone, and display multiple states together that don't necessarily want to be rendered together.

Reply:
Thanks for this, a really great point that I have noted for later. At the moment I've realised that I don't need this but I will return to this in the future.